posthog/hogql_queries/web_analytics/web_overview_pre_aggregated.py

```diff
@@ -0,0 +1,142 @@
+from typing import TYPE_CHECKING
+from datetime import datetime, UTC
+
+from posthog.hogql import ast
+from posthog.hogql.parser import parse_select
+
+if TYPE_CHECKING:
+    from posthog.hogql_queries.web_analytics.web_overview import WebOverviewQueryRunner
+
+
+class WebOverviewPreAggregatedQueryBuilder:
+    # Supported property keys and their mapping to pre-aggregated table fields
+    SUPPORTED_PROPERTIES = {
+        "$host": "host",
+        "$device_type": "device_type",
+    }
+
+    def __init__(self, runner: "WebOverviewQueryRunner") -> None:
+        self.runner = runner
+
+    def can_use_preaggregated_tables(self) -> bool:
```

Comment:
Maybe this should be an `@property` instead? Maybe even cache it

Reply:
Long term, that could be it, yes. But right now, I would much rather have the paths executing to check the flow as much as possible.